src/seat/pointer/mod.rs

```diff
@@ -341,3 +352,178 @@ where
         }
     }
 }
+
+/// Pointer themeing
+#[derive(Debug)]
+pub struct ThemedPointer {
+    pub(super) themes: Arc<Mutex<Themes>>,
+    pub(super) pointer: wl_pointer::WlPointer,
+    pub(super) current_ptr: String,
+    pub(super) scale: i32,
+}
+
+impl ThemedPointer {
+    pub fn set_cursor(
+        &mut self,
+        conn: &Connection,
+        name: &str,
+        shm: &wl_shm::WlShm,
+        surface: &wl_surface::WlSurface,
+    ) -> Result<(), PointerThemeError> {
+        self.current_ptr = name.into();
+        self.update_cursor(conn, shm, surface)
+    }
+
+    pub fn pointer(&self) -> &wl_pointer::WlPointer {
+        &self.pointer
+    }
+}
+
+/// Specifies which cursor theme should be used by the theme manager.
+#[derive(Debug)]
+pub enum ThemeSpec<'a> {
+    /// Use this specific theme with the given base size.
+    Named {
+        /// Name of the cursor theme.
+        name: &'a str,
+
+        /// Base size of the cursor names.
+        ///
+        /// Note this size assumes a scale factor of 1. Cursor image sizes may be multiplied by the base size
+        /// for HiDPI outputs.
+        size: u32,
+    },
+
+    /// Use the system provided theme
+    ///
+    /// In this case SCTK will read the `XCURSOR_THEME` and
+    /// `XCURSOR_SIZE` environment variables to figure out the
+    /// theme to use.
+    System,
```

Comment:
We could make this the `default` value. You'd need to implement Default by hand since the `#[default]` attribute was only stabilized recently (I think 1.64)